### ISSUE TITLE
Type checker: extension member is reported as resolved when object type is clearly not applicable

### ISSUE THREAD

**Issue by auduchinok** [dotnet/fsharp#11658]
```fsharp
open System.Runtime.CompilerServices

[<Extension>]
type Ext =
    [<Extension>]
    static member M(tuple: int * int) = 1

"".M()
```

Expected: `M` is not resolved.

Actual: `M` is resolved but produces type error:

<img width="320" alt="Screenshot 2021-06-09 at 18 19 52" src="https://user-images.githubusercontent.com/3923587/121382464-4f010b80-c94f-11eb-8cd0-601b5cc32632.png">

**Comment by dsyme** [comment]
This is by design as things stand.  Extension methods are added to the available overloads and overload resolution resolves things. We'd need a language suggestion and RFC to change this.

It migh be tricky to change because overload resolution is sensitive to the number and characteristics of the methods in the overall set and can't  simply be adjusted to pre-filter overloads.

**Comment by auduchinok** [comment]
> It might be tricky to change because overload resolution is sensitive to the number and characteristics of the methods in the overall set and can't simply be adjusted to pre-filter overloads.

Wouldn't such pre-filtering possible at least for cases where extension method is invoked on an clearly incompatible type?

**Comment by dsyme** [comment]
> Sadly no, the existence of additional overloads also defines whether the compiler commits to an overload early or not, influencing type inference.

This is correct, though we could suppress this for such cases.

@auduchinok What are the negative ramifications for the fact the symbol is considered "resolved + type error" rather than "unresolved" - is is mainly problems for tooling? Thanks

**Comment by auduchinok** [comment]
> @auduchinok What are the negative ramifications for the fact the symbol is considered "resolved + type error" rather than "unresolved" - is is mainly problems for tooling?

No, it was about what seemed a wrong method resolution. As a user, I felt an extension for an incompatible type should've been chosen here.

Is it possible to check if a type is compatible the extended one without affecting further type inference? If, say, a type parameter was cloned into some synthetic one and checked against the expected type of the extension method.
